### PR TITLE
Replace HPX_INLINE_CONSTEXPR_VARIABLE with inline constexpr

### PR DESCRIPTION
**Pull request by hkaiser**

### CODE REVIEW

File: libs/core/config/include/hpx/config/constexpr.hpp [hunk 1]
```diff
@@ -33,7 +33,7 @@
 #if defined(HPX_COMPUTE_DEVICE_CODE)
```
Comment (msimberg) [inline libs/core/config/include/hpx/config/constexpr.hpp:33]: Could you also remove the macro itself a few lines up?

**Review by msimberg — CHANGES_REQUESTED** [review]
Looks good to me! I'd just remove the macro itself as well.

File: libs/core/config/include/hpx/config/constexpr.hpp (hunk 1 above)
Comment (hkaiser) [reply libs/core/config/include/hpx/config/constexpr.hpp:33]: Good catch, will do.

**Review by aurianer — APPROVED** [review]
LGTM thanks!

**Comment by msimberg** [comment]
Unrelated to this PR, but the partitioned vector minmax failure is unfortunately not gone: https://cdash.cscs.ch/testDetails.php?test=57780895&build=198648.

**Comment by msimberg** [comment]
bors merge

**Review by msimberg — APPROVED** [review]

**Comment by msimberg** [comment]
bors merge

**Comment by msimberg** [comment]
If bors ends up being more annoying than useful we can revise the conditions it requires for a merge. However, I'd try it like this still for a while. At least this forces someone to check that requests for changes have actually been resolved.

**Pull request merged**